components/compositing/windowing.rs

```diff
@@ -103,8 +112,10 @@ pub enum EmbedderEvent {
     /// Sent when the user triggers a media action through the UA exposed media UI
     /// (play, pause, seek, etc.).
     MediaSessionAction(MediaSessionActionType),
-    /// The visibility of the webview has changed.
-    WebViewVisibilityChanged(TopLevelBrowsingContextId, bool),
+    /// Mark webview as invisible due to external factors, regardless of whether it is being painted.
+    MarkWebViewInvisible(TopLevelBrowsingContextId),
+    /// Unmark webview as invisible due to external factors. The webview may remain invisible if it's not being painted.
+    UnmarkWebViewInvisible(TopLevelBrowsingContextId),
```

Comment:
Let's do a single message here `UpdateWebViewVisibility` with an enum argument.

Reply:
the name UpdateWebViewVisibility may mislead consumers into believing that (true) and (false) will show and hide the webview, whereas what we’re actually doing is notifying the compositor that the webview is (or is no longer) invisible due to external factors. we tried to hint at this more passive and specific meaning with “Mark” and “Unmark”, but maybe there’s a better way?